### ISSUE TITLE
"parking lot" should find `amenity=parking` also in UK English

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/id-tagging-schema#461]
(I may implement it, after https://github.com/openstreetmap/id-tagging-schema/pull/337 is reviewed/rejected/merged)

EDIT: asked on osm IRC "Also, I have UK English question. I know that "parking lot" is not primary term for amenity=parking in UK English. But would it be reasonable to list is among options if user searches for it (and they have set UK English as a language)?"

EDIT:

<img width="386" height="561" alt="Image" src="https://github.com/user-attachments/assets/a88bc6d3-a81d-4b17-a548-5c356107de35" />

**Comment by Hufkratzer** [comment]
How do you intend to do that? In default language "en" It is already found:

![parking_lot](https://user-images.githubusercontent.com/5958184/168814868-5cff71a4-fc18-4e5b-856b-f512a3dd5e47.jpg)

Only in the language "en_GB" it is not found, because it was translated as "Car Park".

**Comment by matkoniecz** [comment]
> How do you intend to do that?

Add an alias like in say https://github.com/openstreetmap/id-tagging-schema/pull/335

**Comment by tyrasd** [comment]
I guess this could probably be fixed in iD's code directly because this is likely an issue with all presets whose name has been translated in any (non US) dialect of English: Any preset _name_ should also be automatically be included in the search term for all English locales. This would be the only "good" solution if you ask me, because otherwise we would potentially need to include every preset's `name` in their respective list of `terms` (see the [readme section](https://github.com/openstreetmap/id-tagging-schema#translations) for context).

**Comment by matkoniecz** [comment]
Such systematic idea is likely superior than doing it with all terms as it is being spotted. Though I am not going to promise implementing this wider scope.

**Comment by westnordost** [comment]
Well, another idea here: Instead of always also matching against the en-US default localization, instead match against the **tag value**. E.g. if you type "parking", it will find `amenity=parking` regardless of the user's locale because that's the OSM tag value of the preset.

**Comment by matkoniecz** [comment]
Would it be possible to add this English dialect aliases to output files?

Note that if such aliasing will be done on iD side it would force StreetComplete and GoMap!! and Every Door by Zverik and so on to implement the same mechanism.

For example see https://github.com/westnordost/osmfeatures/issues/13 which appears to be result of the same issue.

`alcohol shop` finds `shop=alcohol` in en-GB and fails to find it in en-US

https://github.com/openstreetmap/id-tagging-schema/blob/main/dist/translations/en-GB.json#L3453
https://github.com/openstreetmap/id-tagging-schema/blob/main/dist/translations/en.json#L7137

**Comment by matkoniecz** [comment]
Build script seems to be in https://github.com/ideditor/schema-builder/blob/main/lib/build.js - and I could try implementing this as part of improving StreetComplete (and improve iD, GoMap!!, Every Door and maybe also other editors).

**Comment by westnordost** [comment]
@matkoniecz That does not make sense in my opinion (I am the author of osmfeatures library):

The primary reason being that `en.json` is actually `en-US`, i.e. there is no `en-US.json`. Mateusz idea premises on that the translations are organized in a form like
- `pt` - Portuguese translations: contains common translations
- `pt-PT` - Portuguese translations (Portugal dialect)
- `pt-BR` - Portuguese translations (Brazil dialect)

If the translations were organized in that manner, merging `pt` into `pt-PT` for distribution would make sense. But the translations are not organized in that manner. We have 

- `pt` - all Portuguese translations (implicitly Portugal dialect)
- `pt-BR` - all Portuguese translations (implicitly Brazil dialect)

The same for other languages that have significant dialects in different countries, i.e. `en`. So, merging together `pt` and `pt-BR` will then just include the Brazilian words into the localization for Portugal. E.g. `highway=bus_stop` will both be named "Paragem de autocarro" (correct) and "Ponto de ônibus" (but that's Brazilian Portuguese).

In the end, whether to fall back or even merge (in)to another locale should remain a client-side decision. I.e. if you decide that merging en-US into en-GB may not be the cleanest solution, but nevertheless it improves things (may want to consult with British users though), then it is a decision you should make for iD and not for any user of this preset data.

**Comment by matkoniecz** [comment]
My premise is that if term is used for something in one dialect of English, then it will be an useful alias in any other dialect of English.

I am aware that `en` is actually `en-US`

Note that it would be only alias: not something shown as a label and only mattering when someone used this term on their own.

Lets take fictional example and say than in `EN-AU` name for parking lot is "foobar". Is it useful to show parking lot when user searched for "foobar" while using `EN-GB`?

How likely is that (1) alias would be used also in other dialects, maybe less commonly (2) someone would be mixing multiple dialects and use terms from different ones at once (especially common with people learning English as a foreign language)?

How likely is that alias would result in actively misleading/confusing/unwanted matches? Like #237?

Summoning @Zelonewolf (hope that it is OK) as I am NOT a native speaker of English.

> E.g. highway=bus_stop will both be named "Paragem de autocarro" (correct) and "Ponto de ônibus" (but that's Brazilian Portuguese).

I am not proposing that. I am proposing that it would be named "Paragem de autocarro". But findable also if someone would type "Ponto de ônibus" (or "Ponto" or "ônibus").

**Comment by tyrasd** [comment]
I would propose to only do the following:

* append `en` name of preset to `en-*` terms (except duplicates)
* append `en` aliases of preset to `en-*` terms (except duplicates)

**Comment by ZeLonewolf** [comment]
I don't really have an opinion on how the language dialects are structured.  There are certainly examples of words that mean one thing in en_GB and something else in en_US,  For example en_GB "chips" means en_US "french fries" and en_US "chips" means en_GB "crisps".  I'm not sure how many of these cases would apply to OSM features, however...

**Comment by westnordost** [comment]
_pavement_ comes to mind

**Comment by westnordost** [comment]
Hm well, I just want to advert at the other possible solution that would solve the use case as described as well: https://github.com/openstreetmap/id-tagging-schema/issues/461#issuecomment-1134045651
This would be a feature in iD of course, not in this schema.

**Comment by tyrasd** [comment]
> match against the tag value.

iD already does support this (see https://github.com/openstreetmap/iD/issues/8869#issuecomment-1004306198).

While this helps in some cases (e.g. when searching for _parking_), in this particular case (searching for _parking lot_) it doesn't do the trick.

**Comment by bgo-eiu** [comment]
I agree with using overlapping terms/aliases for all other English dialects if you have one locale selected. Not everyone I know even speaks the same dialect of English so sometimes I forget which ones might be locale specific.

The tag names themselves are also already a mix of dialects as well

**Comment by ZeLonewolf** [comment]
> _pavement_ comes to mind. That's the usually paved walk for pedestrians at the side of the street in British English. And in American English, it is the main part of the street that has pavement, i.e. often everything _except_ the sidewalk.

In en_US, _pavement_ refers to any paved area.  The street has pavement, the sidewalk has pavement, etc.

**Comment by 1ec5** [comment]
Parking lots are pavement too. 😉

If either the build scripts or individual clients like iD mix terms from different dialects, these additional terms should be weighted much less than terms from the current dialect.

Also, there would already be use cases for language-specific tweaks to the preset search algorithm: https://github.com/openstreetmap/iD/issues/8242#issuecomment-742220871. Maybe any change related to this request could be scoped to just English for now, where the impact would be better understood.

**Comment by RudyTheDev** [comment]
I think this should really be implemented as shared names/aliases between dialects. I doubt the occasional false positives would be worse than missing so many cases as currently. Otherwise, we have to resort to individually reporting cases where a search term does not find something expected/obvious. I have avoided doing this myself, because there are just so many examples, but most cases aren't as bad as completely missing the preset from the list. It's also a lot of effort for such a tiny issue, so I imagine people aren't that keen on reporting these one at a time.

Recently, I was searching for `shop=bag` with "bag shop". I had no idea it's apparently "luggage" on en-GB. And the actual preset even has both "bag" and "luggage" in name too.

Just to illustrate my point:

<img width="234" height="1996" alt="Image" src="https://github.com/user-attachments/assets/5f17bd3a-5c9a-4350-a1ae-eb965f387773" />

I mean, if a new user or even an experienced user is searching for these, they will likely give up long before they reach the actual entry. The only reason I learned about this example is because I manually entered the shop key value and it matched the preset then. (By the way, this is an old screenshot - it currently does not match "bag shop" at all anymore, which is what I was going to report before looking at this issue.)

**Comment by k-yle** [comment]
@andrewharvey as an en-AU translator, do you have any opinions?

I sometimes forget to add the american preset `label` to the en-NZ `terms`, which creates 3 months of annoyance until the next iD release.[^1] So i'd support this. 

There's also so much americani**𝘀**ation nowadays (and just globalisation in general), so it's normal to use words from other dialects


[^1]: for example, searching for "[subway entrance](https://osm.wiki/Tag:railway=subway_entrance)s" finds 0 search results in en-AU at the moment

**Comment by andrewharvey** [comment]
Lots of good points already raised here. I can see how automatically combining the alias terms would improve searchability in many cases and also make maintenance of the terms easier. So I'm in favour of that, ideally the ranking would be lower so matching search terms within your specific locale appear first.

Even native `en-au` or `en-nz` speakers can be influenced by other cultures and pick up terms across the English languages.

Some terms could lead to confusion, but if they are only used as search aliases and the actual preset name shown is still from the choosen locale, I don't think it should be an issue, it shouldn't hurt to have a few extra search results.

For example the `shop=convenience` alias/translation from `en-nz` includes/is "dairy". In `en-au` if I search for dairy I might be looking for a place that sells milkshakes ice cream or something, so as a search term only it's fine.
